frontend/src/app/providers/complaint-filter-provider.tsx

```diff
@@ -52,13 +52,19 @@ const mapFilters = (complaintFilters: Partial<ComplaintFilters>) => {
     natureOfComplaintFilter,
     outcomeAnimalFilter,
   } = complaintFilters;
+
+  // Parse the start and end date filters into Date objects if they exist.
+  const parsedStartDate = startDateFilter ? new Date(startDateFilter) : undefined;
+  // If start date is set and end date is not, default to current date for end date.
+  const parsedEndDate = endDateFilter ? new Date(endDateFilter) : parsedStartDate ? new Date() : undefined;
```

Comment:
Uh oh now it looks like Sonar is going to whine about embedded ternary.   I think you can get around it by splitting it up into.